src/libos/crates/io-uring-callback/src/lib.rs

```diff
@@ -351,6 +351,40 @@ impl IoUring {
         self.push_entry(entry, callback)
     }
 
+    /// Push a timeout request into the submission queue of the io_uring.
+    ///
+    /// # Safety
+    ///
+    /// See the safety section of the `IoUring`.
+    pub unsafe fn timeout(
+        &self,
+        timespec: *const types::Timespec,
+        count: u32,
+        flags: types::TimeoutFlags,
+        callback: impl FnOnce(i32) + Send + 'static,
+    ) -> IoHandle {
+        let entry = opcode::Timeout::new(timespec)
+            .count(count)
+            .flags(flags)
+            .build();
+        self.push_entry(entry, callback)
+    }
+
+    /// Push a timeout_remove request into the submission queue of the io_uring.
+    ///
+    /// # Safety
+    ///
+    /// See the safety section of the `IoUring`.
+    pub unsafe fn timeout_remove(
+        &self,
+        user_data: u64,
+        flags: types::TimeoutFlags,
+        callback: impl FnOnce(i32) + Send + 'static,
+    ) -> IoHandle {
+        let entry = opcode::TimeoutRemove::new(user_data).flags(flags).build();
+        self.push_entry(entry, callback)
+    }
```

Comment:
What is the relationship between cancel and timeout_remove? Can we unify cancel and remove with `IoHandle::cancel`?

Reply:
- **async_cancel:** general cancel (from kernel 5.5)
- **timout_remove:** cancel timeout (from kernel 5.5), update timeout (from kernel 5.11)

In cancel cases, `timeout_remove` is a subset of `async_cancel`, hence we can use `async_cancel` in `IoHanlde::cancel`.